rllib/tests/test_nested_observation_spaces.py

```diff
@@ -120,6 +120,7 @@ def step(self, action):
 
 class NestedMultiAgentEnv(MultiAgentEnv):
     def __init__(self):
```

Comment:
Nit: Can we start making all our example envs' (`MultiAgentEnv` or others) with the c'tor signature:

```
def __init__(self, env_ctx: EnvContext):
    super().__init__()
```

That way, we start being a good role model by showing how to best sub-class any env.

Also, envs defined like this can more easily be configured in  a user's config:

```
config:
    env: HierarchicalWindyMazeEnv  # <- just the class suffices here, no tune.register_env needed!
    env_config:
        # <- c'tor args that go into `env_ctx`
```

Reply:
I can definitely do this for this environment, but for other environments, I suggest we open another issue/pr. This one has already gotten p big.

Reply:
I've opened an issue for it. I have some thoughts about this that I want to take offline.